### PR TITLE
Optimiser update

### PR DESCRIPTION
**Pull request by EdoardoGiancarli**
## **Optimiser**
Update for the **Optimiser** used during the IROS procedure to fit the sources (i) camera local-frame position and (ii) counts values.

- updated source model used in the optimisation;
- updated fit procedure: from `minimize` to `curve_fit` (from the `scipy` lib). Also removed the **loss** metric, since it is computed inside the `curve_fit` pipeline;
- inserted starting fluence correction wrt the camera coding power value [1];
- inserted narrower source local-frame coords boundaries for the fit (from tests we observed that the **true** source position does not deviate so much from the source **peak** position in the sky image --less than 2px--);

[1] Evangelista et al., "Design and performance of the coded mask for the Lunar Electromagnetic Monitor in X-rays (LEM-X)", in preparation (2025)

## **Tests**
The optimiser tests have been commented, at the moment. They will be updated once we decide the final structure for the optimiser procedure.

## **Notes**
With `curve_fit`, the optimisation procedure proved to be less computational expensive wrt `minimize`:
- by a factor ~6 wrt non-cached version (used with source fractional shift)
- by a factor ~3 wrt cached version for old source model

The source position and fluence residues are in the worse case similar to the ones obtained with the `minimize` procedure.
To resume, we improved the optimisation computational cost, while preserving the source parameters fit efficiency.

### CODE REVIEW

File: bloodmoon/optim.py [hunk 1]
```diff
@@ -606,16 +416,16 @@ def optimize(
     arg_sky: tuple[int, int],
     vignetting: bool = True,
     psfy: bool = True,
-    model: Literal["fast", "accurate"] = "fast",
+    verbose: bool = True,
```
Comment (peppedilillo) [inline bloodmoon/optim.py]: set default to False

File: bloodmoon/optim.py [hunk 2]
```diff
@@ -624,52 +434,73 @@ def optimize(
...
-    loss = _Loss(model_shift_flux)
-    results = minimize(
-        lambda args: loss((args[0], args[1], args[2]), sky, arg_sky, camera),
-        x0=np.array((sx_start, sy_start, fluence_start)),
-        method="Nelder-Mead",
+    px_dim_x, px_dim_y = (
+        camera.specs.mask_deltax / camera.upscale_f.x,
+        camera.specs.mask_deltay / camera.upscale_f.y,
+    )
+    camera_coding_power = 0.85
```
Comment (peppedilillo) [inline bloodmoon/optim.py]: does this number depend on the mask? If so it should not be hard coded but either 1. computed from the mask pattern here, 2. red from the mask FITS file

File: bloodmoon/optim.py (hunk 1 above)
Comment (EdoardoGiancarli) [reply bloodmoon/optim.py]: Okie

File: bloodmoon/optim.py (hunk 2 above)
Comment (EdoardoGiancarli) [reply bloodmoon/optim.py]: I agree with the second option, as the coding power is computed from simulations

File: bloodmoon/optim.py [hunk 3]
```diff
@@ -443,7 +438,7 @@ def optimize(
     vignetting: bool = True,
     psfy: bool = True,
     verbose: bool = False,
```
Comment (peppedilillo) [inline bloodmoon/optim.py]: Move verbose to last argument position.

Comment (EdoardoGiancarli) [reply bloodmoon/optim.py]: roger that
